datatypes/src/raster/no_data_grid.rs

```diff
@@ -0,0 +1,172 @@
+use std::ops::Add;
+
+use super::{
+    grid_traits::{ChangeGridBounds, GridShapeAccess},
+    Grid, GridBoundingBox, GridBounds, GridIdx, GridIndexAccess, GridShape, GridShape1D,
+    GridShape2D, GridShape3D, GridSize, GridSpaceToLinearSpace, NoDataValue,
+};
+use crate::{
+    error::{self},
+    raster::GridContains,
+    util::Result,
+};
+use num_traits::AsPrimitive;
+use serde::{Deserialize, Serialize};
+use snafu::ensure;
+
+#[derive(Debug, Eq, PartialEq, Clone, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct NoDataGrid<D, T> {
+    pub shape: D,
+    pub no_data_value: T,
+}
+
+pub type NoDataGrid1D<T> = NoDataGrid<GridShape1D, T>;
+pub type NoDataGrid2D<T> = NoDataGrid<GridShape2D, T>;
+pub type NoDataGrid3D<T> = NoDataGrid<GridShape3D, T>;
+
+impl<D, T> NoDataGrid<D, T>
+where
+    D: GridSize,
+    T: Clone,
```

Comment:
i think it would be nice to have a Grid of Coordinates. So no, i do not think it should be pixel. While i don't see a reason to have a NoDataGRid of something other then pixels it should be the same as for Grid.

Reply:
Maybe then also Copy here, since Coordinates are Copy.

Reply:
24f5725